From: Robert Luberda <robert@debian.org>
Date: Thu, 17 Mar 2011 23:17:51 +0100
Subject: 0027 Include Debian Makefile.languages.inc

This patch adds include ../../debian/local/Makefile.languages.inc
for american/british/english languages and enables overwriting the install
target.

Additionally support for a special `debian-none' value of MASTERHASH
is added into top-level Makefile.
---
 Makefile                    | 2 +-
 languages/american/Makefile | 5 ++++-
 languages/british/Makefile  | 5 ++++-
 languages/english/Makefile  | 6 +++++-
 4 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index f3cc321..09e5707 100644
--- a/Makefile
+++ b/Makefile
@@ -422,7 +422,7 @@ install-languages:
 	    ||  $(MAKE) NEWDIR=$(DESTDIR)$$LIBDIR mkdirpath; \
 	  set -x; \
 	  cd $(DESTDIR)$$LIBDIR; \
-	  if [ $$MASTERHASH != $$DEFHASH ]; then \
+	  if [ $$MASTERHASH != debian-none ] && [ $$MASTERHASH != $$DEFHASH ]; then \
 	    rm -f $$DEFHASH; \
 	    $$LINK -s $$LIBDIR/$$MASTERHASH $$DEFHASH; \
 	  fi
diff --git a/languages/american/Makefile b/languages/american/Makefile
index f58987c..bfe9c91 100644
--- a/languages/american/Makefile
+++ b/languages/american/Makefile
@@ -155,11 +155,14 @@ all:  $(CONFIG)
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+INSTALL_TARGET: install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install:	all $(CONFIG)
+$(INSTALL_TARGET):	all $(CONFIG)
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
diff --git a/languages/british/Makefile b/languages/british/Makefile
index 48f4390..34d740c 100644
--- a/languages/british/Makefile
+++ b/languages/british/Makefile
@@ -155,11 +155,14 @@ all:  $(CONFIG)
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+INSTALL_TARGET := install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install:	all $(CONFIG)
+$(INSTALL_TARGET):	all $(CONFIG)
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
diff --git a/languages/english/Makefile b/languages/english/Makefile
index 2c16aaa..04571da 100644
--- a/languages/english/Makefile
+++ b/languages/english/Makefile
@@ -202,11 +202,15 @@ all:  $(CONFIG)
 	done
 	$(MAKE) SHELLDEBUG=$(SHELLDEBUG) $(HASHFILES)
 
+
+INSTALL_TARGET := install
+include ../../debian/local/Makefile.languages.inc
+
 # Note the fooling around with LIBDIR.  It might be a relative path,
 # relative to the top of the ispell source tree.  So we have to cd to
 # ../.. before referring to $LIBDIR.  There must be a better way...
 # 
-install: all $(CONFIG)
+$(INSTALL_TARGET): all $(CONFIG)
 	@. $(CONFIG); \
 	  set -x; \
 	  cd ../..; \
